src/python/py/models/builder.py

```diff
@@ -2447,6 +2577,38 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
         super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
 
 
+class Phi3MoE128KModel(MistralModel):
+    def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
+        super().__init__(config, io_dtype, onnx_dtype, ep, cache_dir, extra_options)
+        assert io_dtype == TensorProto.FLOAT16, "This model only supports float16 io type."
+
+        self.rotemb_attrs["rotemb_name"] = "rotary_emb"
+
+        self.layernorm_attrs["simple"] = False
+
+        self.moe_attrs["num_experts"] = 16
+        self.moe_attrs["top_k"] = 2
+        self.moe_attrs["activation_type"] = "silu"
+        self.moe_attrs["normalize_routing_weights"] = 0
+        self.moe_attrs["use_sparse_mixer"] = 1
+        self.moe_attrs["use_int4"] = 0 if "use_8bits_moe" in extra_options and extra_options["use_8bits_moe"] == "1" else 1
```

Comment:
```suggestion
        self.moe_attrs["use_int4"] = 0 if "use_8bits_moe" in extra_options else 1
```

Even though the documentation shows the official way to select extra options, users seem to use a wide set of versions to denote that they want to use an extra option (e.g. `use_8bits_moe=1`, `use_8bits_moe=true`, `use_8bits_moe=True`, etc). I think we can just check if the user passed the option or not to understand that they want to utilize it.

Reply:
A corner case is that user may add use_8bits_moe=0 which causes logic flaw. we can do some checks for extra_options